src/state/LogIn/loginSlice.js

```diff
@@ -0,0 +1,45 @@
+import { createSlice } from "@reduxjs/toolkit";
+import { createAsyncThunk } from "@reduxjs/toolkit";
+
+import { privatePost } from "../../utilities";
+export const fetchUserData = createAsyncThunk(
+  "user/signin",
+  async (user, { rejectWithValue }) => {
+    try {
+      const response = await privatePost("/signin", user);
```

Comment:
Login is not private. User will get the token by login. User won't have it before login.